packages/test/test-end-to-end-tests/src/test/serializeAfterFailedAttach.spec.ts

```diff
@@ -95,171 +94,184 @@ describeCompat(
-				loader,
-				request,
-			);
-
-			// create a new data store
-			const peerDataStore = await createPeerDataStore(
-				defaultDataStore.context.containerRuntime,
-			);
-			const dataStore2 = peerDataStore.peerDataStore as TestFluidObject;
-
-			// create a new dds
-			const ddsId = "notbounddds";
-			const dds2 = dataStore2.runtime.createChannel(ddsId, SharedString.getFactory().type);
-
-			// attach the new data store and dds
-			const dds2Key = "dds2";
-			const dataStore2Key = "dataStore2";
-			const rootOfDataStore1 =
-				await defaultDataStore.getSharedObject<ISharedMap>(sharedMapId);
-			rootOfDataStore1.set(dataStore2Key, dataStore2.handle);
-			rootOfDataStore1.set(dds2Key, dds2.handle);
-
-			// serialize and rehydrate
-			const snapshotTree = container.serialize();
-			const rehydratedContainer =
-				await loader.rehydrateDetachedContainerFromSnapshot(snapshotTree);
-
-			const rehydratedEntryPoint =
-				(await rehydratedContainer.getEntryPoint()) as TestFluidObject;
-			const rehydratedRoot =
-				await rehydratedEntryPoint.getSharedObject<ISharedMap>(sharedMapId);
-			const dataStore2Handle: IFluidHandle<TestFluidObject> | undefined =
-				rehydratedRoot.get(dataStore2Key);
-			const dds2Handle: IFluidHandle<ISharedMap> | undefined = rehydratedRoot.get(dds2Key);
-
-			// validate data store
-			assert(dataStore2Handle !== undefined, `handle for [${dataStore2Key}] must exist`);
-			const dataStore2FromRC = await dataStore2Handle.get();
-			assert(dataStore2FromRC, "DataStore2 should have been serialized properly");
-			assert.strictEqual(
-				dataStore2FromRC.runtime.id,
-				dataStore2.runtime.id,
-				"DataStore2 id should match",
-			);
+				// Check for default data store
+				const entryPoint = await rehydratedContainer.getEntryPoint();
+				assert.notStrictEqual(entryPoint, undefined, "Component should exist!");
+				// Might need some other asserts, but not sure what
+			});
+			it(`Can serialize detached container with data stores after failed attach. attachAfterRehydrate: ${attachAfterRehydrate}`, async () => {
+				const provider = getTestObjectProvider();
+
+				// create a detached container and attempt to attach
+				const { container, defaultDataStore } =
+					await createDetachedContainerAndGetEntryPoint(provider);
+
+				// create a new data store
+				const peerDataStore = await createPeerDataStore(
+					defaultDataStore.context.containerRuntime,
+				);
+				const dataStore2 = peerDataStore.peerDataStore as TestFluidObject;
+				const rootOfDataStore1 =
+					await defaultDataStore.getSharedObject<ISharedMap>(sharedMapId);
+				const dataStore2Key = "dataStore2";
+				// attach a new data store
+				rootOfDataStore1.set(dataStore2Key, dataStore2.handle);
+
+				// serialize and rehydrate
+				const snapshotTree = container.serialize();
+
+				const loader = createTestLoader(provider);
+
+				const rehydratedContainer =
+					await loader.rehydrateDetachedContainerFromSnapshot(snapshotTree);
+
```

Comment:
I think you're missing the second attach attempt here, unless we don't want to check attach after rehydrate for data stores.